File: app/src/main/java/com/github/goldy1992/mp3player/client/views/fragments/SearchFragment.java
package com.github.goldy1992.mp3player.client.views.fragments;

import android.app.Activity;
import android.app.SearchManager;
import android.app.SearchableInfo;
import android.content.ComponentName;
import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import android.util.Log;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.view.inputmethod.InputMethodManager;
import android.widget.LinearLayout;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.annotation.VisibleForTesting;
import androidx.appcompat.widget.SearchView;
import androidx.fragment.app.Fragment;

import com.github.goldy1992.mp3player.LogTagger;
import com.github.goldy1992.mp3player.R;
import com.github.goldy1992.mp3player.client.activities.SearchResultActivityInjector;

public class SearchFragment extends Fragment implements LogTagger {

    private LinearLayout linearLayout;
    private SearchView searchView;
    private InputMethodManager inputMethodManager;

    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container,
                             @Nullable Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        return inflater.inflate(R.layout.fragment_search, container, false);

    }

    @Override
    public void onViewCreated(@NonNull View view, Bundle bundle) {

        this.searchView = view.findViewById(R.id.search_view);
        this.inputMethodManager = (InputMethodManager) getContext().getSystemService(Activity.INPUT_METHOD_SERVICE);

         // Get the SearchView and set the searchable configuration
        SearchManager searchManager = (SearchManager) getActivity().getSystemService(Context.SEARCH_SERVICE);
        ComponentName componentName = new ComponentName(getContext(), SearchResultActivityInjector.class);
        SearchableInfo searchableInfo = searchManager.getSearchableInfo(componentName);
        // Assumes current activity is the searchable activity
        searchView.setSearchableInfo(searchableInfo);
        searchView.setIconifiedByDefault(false); // Do not iconify the widget; expand it by default
        searchView.setSubmitButtonEnabled(true);
        searchView.setBackgroundColor(Color.WHITE);


        this.linearLayout = view.findViewById(R.id.search_fragment_layout);
        Drawable background = this.linearLayout.getBackground();
        background.setAlpha(200);
        linearLayout.setOnClickListener(this::onClickOnLayout);

        searchView.setOnClickListener((View v) -> {
            Log.i(getLogTag(), "hit search view");
        });

        searchView.requestFocusFromTouch();
        inputMethodManager.toggleSoftInput(InputMethodManager.SHOW_FORCED, InputMethodManager.HIDE_IMPLICIT_ONLY);

        searchView.setOnQueryTextFocusChangeListener(this::onFocusChange);
    }

    @VisibleForTesting
    public void onClickOnLayout(View view) {
        Log.i(getLogTag(), "hit on click listener");
        inputMethodManager.hideSoftInputFromWindow(getView().getWindowToken(), 0 );
        getFragmentManager().popBackStack();
    }


    @VisibleForTesting
    public void onFocusChange(View v, boolean queryTextFocused) {
        Log.i("tag", "focus changed: has focus: " + queryTextFocused);
        if (!queryTextFocused) {
            getFragmentManager().popBackStack();
        }
    }
    @Override
    public String getLogTag() {
        return "SRCH_FRAGMENT";
    }
}

File: client/src/main/java/com/github/goldy1992/mp3player/client/views/fragments/SearchFragment.kt
package com.github.goldy1992.mp3player.client.views.fragments

import android.app.Activity
import android.app.SearchManager
import android.content.ComponentName
import android.content.Context
import android.graphics.Color
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.View.OnFocusChangeListener
import android.view.ViewGroup
import android.view.inputmethod.InputMethodManager
import androidx.annotation.VisibleForTesting
import com.github.goldy1992.mp3player.client.R
import com.github.goldy1992.mp3player.client.callbacks.Listener
import com.github.goldy1992.mp3player.commons.ComponentClassMapper
import com.github.goldy1992.mp3player.commons.LogTagger
import kotlinx.android.synthetic.main.fragment_search.*
import java.util.*
import javax.inject.Inject

class SearchFragment : MediaFragment(), LogTagger {

    private var inputMethodManager: InputMethodManager? = null

    @Inject
    lateinit var componentClassMapper : ComponentClassMapper

    override fun mediaControllerListeners(): Set<Listener> {
        return Collections.emptySet()
    }

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?,
                              savedInstanceState: Bundle?): View? {

        super.onCreateView(inflater, container, savedInstanceState)
        return inflater.inflate(R.layout.fragment_search, container, false)
    }

    override fun onViewCreated(view: View, bundle: Bundle?) {
        inputMethodManager = requireContext().getSystemService(Activity.INPUT_METHOD_SERVICE) as InputMethodManager
        // Get the SearchView and set the searchable configuration
        val searchManager = requireActivity().getSystemService(Context.SEARCH_SERVICE) as SearchManager
        val componentName = ComponentName(requireContext(), componentClassMapper.searchResultActivity!!)
        val searchableInfo = searchManager.getSearchableInfo(componentName)
        // Assumes current activity is the searchable activity
        searchView.setSearchableInfo(searchableInfo)
        searchView.setIconifiedByDefault(false) // Do not iconify the widget; expand it by default
        searchView.isSubmitButtonEnabled = true
        searchView.setBackgroundColor(Color.WHITE)
        val background = linearLayout.getBackground()
        background.alpha = 200
        linearLayout.setOnClickListener(View.OnClickListener { onClickOnLayout() })
        searchView.setOnClickListener(View.OnClickListener { Log.i(logTag(), "hit search view") })
        searchView.requestFocusFromTouch()
        inputMethodManager!!.toggleSoftInput(InputMethodManager.SHOW_FORCED, InputMethodManager.HIDE_IMPLICIT_ONLY)
        searchView.setOnQueryTextFocusChangeListener(OnFocusChangeListener { _: View?, queryTextFocused: Boolean -> onFocusChange(queryTextFocused) })
    }

    @VisibleForTesting
    fun onClickOnLayout() {
        Log.i(logTag(), "hit on click listener")
        inputMethodManager!!.hideSoftInputFromWindow(requireView().windowToken, 0)
        this.parentFragmentManager.popBackStack()
    }

    @VisibleForTesting
    fun onFocusChange(queryTextFocused: Boolean) {
        Log.i("tag", "focus changed: has focus: $queryTextFocused")
        if (!queryTextFocused) {
            val fragmentManager = parentFragmentManager
            if (!fragmentManager.isDestroyed) {
                fragmentManager.popBackStack()
            }
        }
    }

    override fun logTag(): String {
        return "SRCH_FRAGMENT"
    }

    override fun initialiseDependencies() {
createMediaFragmentSubcomponent()
        ?.inject(this)
    }
}